src/argilla/training/autotrain_advanced.py

```diff
@@ -0,0 +1,260 @@
+                params = Params(
+                    task=self.task,
+                    training_type=training_type,
+                ).get()
+                for key, value in params.items():
+                    if key not in set_params:
+                        set_params[key] = value.DEFAULT
+
+                self.trainer_kwargs[training_type][idx] = set_params
+
+    def update_config(self, **kwargs):
+        """
+        Updates the `setfit_model_kwargs` and `setfit_trainer_kwargs` dictionaries with the keyword
+        arguments passed to the `update_config` function.
+        """
+        self._model = kwargs.get("model", self._model)
+        if self._model.lower() == "autotrain":
+            self._model = "autotrain"
+        if "hub_model" in kwargs:
+            if not isinstance(kwargs["hub_model"], list):
+                raise ValueError("hub_model must be a list of dictionaries.")
+            self.trainer_kwargs["hub_model"] = kwargs["hub_model"]
+        if "autotrain" in kwargs:
+            if not isinstance(kwargs["autotrain"], list):
+                raise ValueError("autotrain must be a list of dictionaries.")
+            self.trainer_kwargs["autotrain"] = kwargs["autotrain"]
+
+        self.init_training_args()
+        self.initialize_project()
+
+    def __repr__(self):
+        formatted_string = [
+            "Choose EITHER `autotrain` OR a model from the hub `hub_model` as main key to update parameters.",
+            f"model: {self._model}",
+        ]
+        for arg_dict_key, arg_dict_single in self.trainer_kwargs.items():
+            arg_dict_key += ": List[dict]"
+            formatted_string.append(arg_dict_key)
+            for idx, item in enumerate(arg_dict_single):
+                for key, val in item.items():
+                    formatted_string.append(f"\tjob{idx+1}-{key}: {val}")
+        return "\n".join(formatted_string)
+
+    def train(self, output_dir: str):
+        """
+        We create a SetFitModel object from a pretrained model, then create a SetFitTrainer object with
+        the model, and then train the model
+        """
+        project_id = self.project.create()
+        self.project.approve(project_id)
+
+    def init_model(self, new: bool = False):
+        pass
+
+    def init_pipeline(self):
+        pass
+
+    def predict(self, text: Union[List[str], str], as_argilla_records: bool = True, **kwargs):
+        """
+        The function takes in a list of strings and returns a list of predictions
+
+        Args:
+          text (Union[List[str], str]): The text to be classified.
+          as_argilla_records (bool): If True, the prediction will be returned as an Argilla record. If
+        False, the prediction will be returned as a string. Defaults to True
+
+        Returns:
+          A list of predictions
+        """
+        self._logger.error("Use framework=`transformers` for inference.")
```

Comment:
As we have the information to do so, I'd show here an example on how to load the trained model/s using `transformers`, just to serve as a tip